js/src/attribute-mapping/attribute-mapping-rule-modal.js

```diff
@@ -26,13 +27,45 @@ import AttributeMappingCategoryControl from '.~/attribute-mapping/attribute-mapp
  * @param {Function} [props.onRequestClose] Callback on closing the modal
  */
 const AttributeMappingRuleModal = ( { rule, onRequestClose = noop } ) => {
-	const [ selectedAttribute, setSelectedAttribute ] = useState();
+	const [ selectedAttribute, setSelectedAttribute ] = useState( '' );
 	const [ dropdownVisible, setDropdownVisible ] = useState( false );
 
 	const { data: attributes } = useMappingAttributes();
-	const { data: sources } = useMappingAttributesSources( selectedAttribute );
-	const isEnum = attributes?.find( ( { id } ) => id === selectedAttribute )
-		?.enum;
+	const {
+		data: sources = {},
+		hasFinishedResolution: sourcesHasFinishedResolution,
+	} = useMappingAttributesSources( selectedAttribute );
+
+	const isEnum =
+		attributes.find( ( { id } ) => id === selectedAttribute )?.enum ||
+		false;
+
+	const sourcesOptions = [
+		// Todo: Check this in the future. (Due to an error on my side returning object in the backend)
+		...Object.keys( sources ).map( ( sourceKey ) => {
```

Comment:
Wonder if there will be other options to be added together with the array literal of `sourcesOptions`? If not, it could be replaced with `const sourcesOptions = Object.keys( sources ).map // ...` without spreading the mapped results again.